psync-fconsumer/full-consumer.h

```diff
@@ -0,0 +1,55 @@
+#include <stdio.h>
+#include <netdb.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <ndn-lite.h>
+#include "ndn-lite/encode/name.h"
+#include "ndn-lite/encode/data.h"
+#include "ndn-lite/encode/interest.h"
+#include <time.h>
+#include <stdlib.h>
+
+
+#ifndef FULL_CONSUMER_H
+#define FULL_CONSUMER_H
+
+
+typedef struct 
+{   
+    // Store sequence number for the prefix.
+    ndn_name_t prefix;
+    uint64_t sequence_number;
+} prefixes;
+
+typedef struct
+{
+// define all the required variables i.e. face, io-service and all
+
+    ndn_unix_face_t* unix_face;
```

Comment:
Thanks for your interest of contributing! But NDN-Lite doesn't have any "actual" faces but a base class and a dummy face for test use. 

If you want to use UNIX face, I wonder the [POSIX package](https://github.com/named-data-iot/ndn-iot-package-over-posix) is more appropriate for you to contribute to? Also  the `psync-fconsumer` looks like a consumer for PSync, which is a **protocol** and related files should be put into `app-support`.

Reply:
PSync protocol shouldn't depend on particular face type. It only needs a pointer to the base type.